aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2739,12 +2782,19 @@ impl VMValidator for AptosVM {
             .features()
             .is_enabled(FeatureFlag::ALLOW_SERIALIZED_SCRIPT_ARGS)
         {
-            if let TransactionPayload::Script(script) = transaction.payload() {
-                for arg in script.args() {
-                    if let TransactionArgument::Serialized(_) = arg {
-                        return VMValidatorResult::error(StatusCode::FEATURE_UNDER_GATING);
+            match transaction.payload() {
```

Comment:
it's probably better to have txn.payload() -> Option\<Script\>

Reply:
Done.